x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -552,6 +567,15 @@ protected static String originAttributes(ThreadContext threadContext, TransportM
         return restOriginTag(threadContext).orElse(transportOriginTag(message).orElse(localNodeInfo.localOriginTag));
     }
 
+    private String opaqueId() {
+        String opaqueId = threadContext.getHeader(Task.X_OPAQUE_ID);
+        if (opaqueId != null) {
+            return opaqueId;
+        } else {
+            return "";
+        }
+    }
+
```

Comment:
Generally we don't output empty brackets `<field>=[]`, eg. `indices`, so `opaque_id=[]` is better to not being displayed at all.

Reply:
@albertzaharovits are you saying that you prefer `opaque_id=[]` over the field being left out completely?

Reply:
@jaymode no. the opposite. I prefer not outputing fields with empty values. This is the norm now, and outputing empty field values is only useful in a tabular log format (column names at the top).

Reply:
Makes sense. I will change it in a bit.